src/States/MapViewStateDraw.cpp

```diff
@@ -69,49 +69,60 @@ void MapViewState::drawMiniMap()
 	renderer.drawImage(isHeightmapToggled ? mHeightMap : mMapDisplay, miniMapBoxFloat.startPoint());
 
-	if (ccLocationX() != 0 && ccLocationY() != 0)
+	const auto miniMapOffset = mMiniMapBoundingBox.startPoint() - NAS2D::Point<int>{0, 0};
+	const auto ccPosition = ccLocation();
+	if (ccPosition != NAS2D::Point<int>{0, 0})
 	{
-		renderer.drawSubImage(mUiIcons, ccLocationX() + mMiniMapBoundingBox.x() - 15, ccLocationY() + mMiniMapBoundingBox.y() - 15, 166, 226, 30, 30);
-		renderer.drawBoxFilled(ccLocationX() + mMiniMapBoundingBox.x() - 1, ccLocationY() + mMiniMapBoundingBox.y() - 1, 3, 3, 255, 255, 255);
+		const auto ccOffsetPosition = ccPosition + miniMapOffset;
+		const auto ccCommRangeImageRect = NAS2D::Rectangle<int>{166, 226, 30, 30};
+		renderer.drawSubImage(mUiIcons, ccOffsetPosition - ccCommRangeImageRect.size() / 2, ccCommRangeImageRect);
+		renderer.drawBoxFilled(NAS2D::Rectangle<int>::Create(ccOffsetPosition - NAS2D::Vector<int>{1, 1}, NAS2D::Vector<int>{3, 3}), NAS2D::Color::White);
 	}
 
-	for (auto _tower : Utility<StructureManager>::get().structureList(Structure::CLASS_COMM))
+	auto& structureManager = Utility<StructureManager>::get();
+	for (auto commTower : structureManager.structureList(Structure::CLASS_COMM))
 	{
-		if (_tower->operational())
+		if (commTower->operational())
 		{
-			Tile* t = Utility<StructureManager>::get().tileFromStructure(_tower);
-			renderer.drawSubImage(mUiIcons, t->x() + mMiniMapBoundingBox.x() - 10, t->y() + mMiniMapBoundingBox.y() - 10, 146, 236, 20, 20);
+			const auto commTowerPosition = structureManager.tileFromStructure(commTower)->position();
+			const auto commTowerRangeImageRect = NAS2D::Rectangle<int>{146, 236, 20, 20};
+			renderer.drawSubImage(mUiIcons, commTowerPosition + miniMapOffset - commTowerRangeImageRect.size() / 2, commTowerRangeImageRect);
 		}
 	}
 
-	for (auto _mine : mTileMap->mineLocations())
+	for (auto minePosition : mTileMap->mineLocations())
 	{
-		Mine* mine = mTileMap->getTile(_mine, 0)->mine();
+		Mine* mine = mTileMap->getTile(minePosition, 0)->mine();
 		if (!mine) { break; } // avoids potential race condition where a mine is destroyed during an updated cycle.
 
-		float mineBeaconStatusOffsetX = 0.0f;
-		if (!mine->active()) { mineBeaconStatusOffsetX = 0.0f; }
-		else if (!mine->exhausted()) { mineBeaconStatusOffsetX = 8.0f; }
-		else { mineBeaconStatusOffsetX = 16.0f; }
+		auto mineBeaconStatusOffsetX = 0;
+		if (!mine->active()) { mineBeaconStatusOffsetX = 0; }
+		else if (!mine->exhausted()) { mineBeaconStatusOffsetX = 8; }
+		else { mineBeaconStatusOffsetX = 16; }
 
-		renderer.drawSubImage(mUiIcons, _mine.x() + mMiniMapBoundingBox.x() - 2, _mine.y() + mMiniMapBoundingBox.y() - 2, mineBeaconStatusOffsetX, 0.0f, 7.0f, 7.0f);
+		const auto mineImageRect = NAS2D::Rectangle<int>{mineBeaconStatusOffsetX, 0, 7, 7};
+		renderer.drawSubImage(mUiIcons, minePosition + miniMapOffset - NAS2D::Vector<int>{2, 2}, mineImageRect);
```

Comment:
More functions for reducing cognitive load:
```cpp
for (const auto& minePosition : mTileMap->mineLocations())
{
    DrawMinimapMineAt(minePosition);
}
```
And possibly the above suggestion inside:
```cpp
DrawMinimapMines() const;
```